lib/compilers/win32.ts

```diff
@@ -38,59 +41,67 @@ export class Win32Compiler extends BaseCompiler {
         return 'win32';
     }
 
-    constructor(compilerInfo, env) {
+    binaryAsmParser: AsmParser;
+
+    constructor(compilerInfo: CompilerInfo & Record<string, any>, env) {
         super(compilerInfo, env);
 
         this.binaryAsmParser = new AsmParser(this.compilerProps);
     }
 
-    newTempDir() {
-        return new Promise((resolve, reject) => {
+    override newTempDir() {
+        return new Promise<string>((resolve, reject) => {
             temp.mkdir({prefix: 'compiler-explorer-compiler', dir: process.env.TMP}, (err, dirPath) => {
                 if (err) reject(`Unable to open temp file: ${err}`);
                 else resolve(dirPath);
             });
         });
     }
 
-    getExecutableFilename(dirPath, outputFilebase) {
+    override getExecutableFilename(dirPath: string, outputFilebase: string) {
         return this.getOutputFilename(dirPath, outputFilebase) + '.exe';
     }
 
-    getObjdumpOutputFilename(defaultOutputFilename) {
+    override getObjdumpOutputFilename(defaultOutputFilename: string) {
         return this.getExecutableFilename(path.dirname(defaultOutputFilename), 'output');
     }
 
-    getSharedLibraryPathsAsArguments(libraries) {
+    override getSharedLibraryPathsAsArguments(libraries) {
         const libPathFlag = this.compiler.libpathFlag || '/LIBPATH:';
 
         return this.getSharedLibraryPaths(libraries).map(path => libPathFlag + path);
     }
 
-    getSharedLibraryLinks(libraries) {
+    override getSharedLibraryLinks(libraries) {
         return _.flatten(
-            _.map(libraries, selectedLib => {
-                const foundVersion = this.findLibVersion(selectedLib);
-                if (!foundVersion) return false;
-
-                return _.map(foundVersion.liblink, lib => {
-                    if (lib) {
-                        return '"' + lib + '.lib"';
-                    } else {
-                        return false;
```

Comment:
TS was very not happy about false being returned here and above, rewrote as a filter/map. Is that appropriate or do we expect to return an array with false entries among the strings?

Reply:
I think this is a better phrasing; we were a bit fast and loose with `false`/`undefined`/`null`/whatever :) @partouf for viz

Reply:
Sure, the returned array should be clean of invalid things.

I don't really like `findLibVersion()` being called twice though. Perhaps you can map first to an intermediate type `{selectedLib, foundVersion}` and filter everything where `foundVersion` is false?

Reply:
.. and then map again I guess :/

Reply:
it's hardly ever going to be more than 5 libraries, so it's probably not going to slow things down

Reply:
Good idea, I'll do that